### DISCUSSION TITLE
Unable to configure azure blob for airflow logs on 2.0.0

### DISCUSSION THREAD

**Discussion by msinha251** [apache/airflow#14425] [Q&A]
Hi All,

I am trying to configure the azure blob for airflow logs [Airflow deployed on AKS], but unable to do so.
I have used helm chart for deployment for airflow deployment on AKS and then followed this [https://airflow.apache.org/docs/apache-airflow/1.10.3/howto/write-logs.html#write-logs-azure](url) for configuring it. 
But it's unable to connect with blob.... Keep getting below error, can someone please suggest or help on this.

`("'NoneType' object is not callable",)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.6/site-packages/celery/app/trace.py", line 412, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/celery/app/trace.py", line 704, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 87, in execute_command
    _execute_in_fork(command_to_exec)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/executors/celery_executor.py", line 120, in _execute_in_fork
    logging.shutdown()
  File "/usr/local/lib/python3.6/logging/__init__.py", line 1946, in shutdown
    h.close()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/microsoft/azure/log/wasb_task_handler.py", line 97, in close
    self.wasb_write(log, remote_loc, append=True)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/microsoft/azure/log/wasb_task_handler.py", line 181, in wasb_write
    self.hook.load_string(
  File "/home/airflow/.local/lib/python3.6/site-packages/cached_property.py", line 36, in __get__
    value = obj.__dict__[self.func.__name__] = self.func(obj)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/microsoft/azure/log/wasb_task_handler.py", line 61, in hook
    return WasbHook(remote_conn_id)
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/microsoft/azure/hooks/wasb.py", line 65, in __init__
    self.connection = self.get_conn()
  File "/home/airflow/.local/lib/python3.6/site-packages/airflow/providers/microsoft/azure/hooks/wasb.py", line 71, in get_conn
    return BlockBlobService(account_name=conn.login, account_key=conn.password, **service_options)
TypeError: 'NoneType' object is not callable`

**Comment by ephraimbuddy** [comment]
Hi @msinha251 , for remote logging to work, you need to follow the documentation which is in https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/logging.html

In summary, you need to update the following in airflow configuration:
```
[logging]
remote_logging=True
remote_base_log_folder=wasb://container-name-you-desire
remote_log_conn_id=wasb_default
```
Create the `wasb_default` connection with the necessary information.
You can also add your logging configuration class:
```
logging_config_class=path.to.your.logging_config

**Reply by msinha251** [reply]
Thanks @ephraimbuddy for quick response. I have followed this for configuring it in same way you mentioned above instead of `remote_base_log_folder=wasb://container-name-you-desire`, I have used `remote_base_log_folder=wasb-container-name-you-desire` as per documentation [https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/logging.html](url)

For your ref below are the code snippets from my airflow env:
    `logging_level: DEBUG`
    `remote_logging: 'True'`
    `remote_log_conn_id: 'azure-blob'`
   `remote_base_log_folder: 'wasb-airflow-logs'`
   `logging_config_class: 'log_config.DEFAULT_LOGGING_CONFIG'`

Copied `airflow/config_templates/airflow_local_settings.py` file in `config/log_config.py`

And also added the connection in airflow web ui with storage account name and key.

Can you please suggest what else is left or should be done to properly configure it ?

Mahesh

**Reply by ephraimbuddy** [reply]
From what I can see, the problem is in your connection.
Update your connection correctly and you won't have any issue.
id: azure-blob
type: azure-blob-storage
login: your-storage-account-name
password: your-storage-account-key

![Screenshot from 2021-02-24 22-12-42](https://user-images.githubusercontent.com/4122866/109066860-fda97500-76ed-11eb-98c3-8dc2566cf50f.png)

**Reply by msinha251** [reply]
Yeah I have created connection from airflow web ui and passed the service account name and respective key as you suggested above or in documentation, even tried with SAS token in extra field. But it's didn't work yet.

Not sure what is wrong in azure blob configuration although able to configure the S3 but can't make it run with Azure blob storage for airflow logs.

Not sure below logs will help but sharing it again:

It's keep on getting the `'NoneType' object is not callable` in worker logs and below lines on airflow web ui:
`*** Log file does not exist: ...`
`*** Fetching from: ...`
`*** Failed to fetch log file from worker.... Failed to establish a new connection: [Errno -2] Name or service not known',))`

**Reply by ephraimbuddy** [reply]
What version of apache-airflow-providers-microsoft-azure are you using

**Reply by msinha251** [reply]
apache-airflow-providers-microsoft-azure==1.0.0

**Reply by ephraimbuddy** [reply]
You need a sas token in extra for it to work. Otherwise, upgrade to the latest version of the apache-airflow-providers-microsoft-azure and you won't need sas token. However, be aware there's a bug in the latest version that makes the log display everything in one line. The bug has been fixed and will be out in the next version of the provider

**Reply by msinha251** [reply]
I have tried adding the sas token in extra field it didn't work. Other then I have also tried with upgraded version of  `apache-airflow-providers-microsoft-azure` but it's same error.

**Reply by ephraimbuddy** [reply]
Run this:
`pip uninstall azure-storage --yes && pip install azure-storage-blob apache-airflow-providers-microsoft-azure==1.1.0`

then try the logging again.

**Reply by msinha251** [reply]
Tried this, but this leads to import issue:
`from azure.storage.blob import BlobClient, BlobServiceClient, ContainerClient, StorageStreamDownloader
ImportError: cannot import name 'BlobClient'`

**Reply by ephraimbuddy** [reply]
It's a known issue and this is how you can resolve it:
`pip uninstall azure-storage apache-airflow-providers-microsoft-azure`
`pip install azure-storage-blob apache-airflow-providers-microsoft-azure==1.1.0`
Please run the commands one after the other

**Reply by ephraimbuddy** [reply]
Just make sure azure-storage is uninstalled and then install azure-storage-blob
